test/geotiff.spec.js

```diff
@@ -1287,6 +1287,33 @@ describe('writeTests', () => {
     expect(normalize(fileDirectory.StripByteCounts)).to.equal(normalize(metadata.StripByteCounts));
     expect(fileDirectory.GDAL_NODATA).to.equal('0\u0000');
   });
+
+  it('should write and read back GeoAsciiParams/GeoDoubleParams keys', async () => {
+    const width = 2;
+    const height = 2;
+    const data = new Float32Array(width * height).fill(1);
+
+    const metadata = {
+      width,
+      height,
+      GeographicTypeGeoKey: 4326,
+      GTModelTypeGeoKey: 2,
+      GeogSemiMajorAxisGeoKey: 6378137.0, // DOUBLE
+      GeogInvFlatteningGeoKey: 298.257223563, // DOUBLE
+      GeogCitationGeoKey: 'WGS 84', // ASCII
+      PCSCitationGeoKey: 'test-ascii', // ASCII
+    };
+
+    const buffer = await writeArrayBuffer(data, metadata);
+    const tiff = await fromArrayBuffer(buffer);
+    const image = await tiff.getImage();
+
+    const geoKeys = image.getGeoKeys();
+    expect(geoKeys.GeogSemiMajorAxisGeoKey).to.be.closeTo(6378137.0, 0.000001);
```

Comment:
@jo-chemla , could you provide a little more background on why this was necessary?  I'm assuming it's a floating-point arithmetic thing, but would love to know a little more details.  What part of the code do you think is causing this?  Thank you!

* on somewhat personal note: I've spent more time than I care to admit dealing with floating point arithmetic issues in the geo space, so I prefer it there's a way to preserves numbers while round-tripping that would be awesome!!